feat: enable using the keyboard to type in accents

nobe4 opened the pull request:
Fix #7

nobe4 committed:
feat: enable using the keyboard to type in accents

nobe4 committed:
fix(held): don't allow rolling over potentially held keys

Without this condition, pressing qw quickly would output #w, like q was
held. This is never what I want, I want a pause for any hold to take
effect.

nobe4 committed:
fix(accents): create a new layout without TD

Using existing TD(...) doesn't play well with leader key, so at the same
time we're moving into leader sequence, we're activating the accent
layout, which only contains characters germane to the various accented
sequences.

This, notably, fixes the `<leader>c c` => `ç` that didn't trigger
before.

nobe4 committed:
feat(nix): add entr as a dependency

nobe4 committed:
feat(deven): move scripts from  makefile

nobe4 committed:
feat(devenv): move watch out of the scripts

tebriel commented on devenv.nix:
```diff
@@ -1,7 +1,36 @@
+      ${pkgs.git}/bin/git ls-files | \
+          entr -c -r CONVERT_TO=rp2040_ce ${pkgs.qmk}/bin/qmk flash -kb ferris/sweep -km nobe4
+    '';
+  };
```
```suggestion
  scripts = let
    qmkExe = lib.getExe pkgs.qmk;
    gitExe = lib.getExe pkgs.git;
  in {
    clean.exec = "rm -rf ./qmk";

    install.exec = ''
      ${qmkExe} config user.qmk_home=$(pwd)/qmk
      ${qmkExe} setup -y

      ln -s $(pwd)/planck $(pwd)/qmk/keyboards/planck/keymaps/nobe4
      ln -s $(pwd)/ferris $(pwd)/qmk/keyboards/ferris/keymaps/nobe4
    '';

    rev5.exec = ''
      ${gitExe} ls-files | \
          entr -c -r ${qmkExe} flash -kb planck/rev5 -km nobe4
    '';

    rev6.exec = ''
      ${gitExe} ls-files | \
          entr -c -r ${qmkExe} flash -kb planck/rev6 -km nobe4
    '';

    ferris.exec = ''
      ${gitExe} ls-files | \
          entr -c -r CONVERT_TO=rp2040_ce ${qmkExe} flash -kb ferris/sweep -km nobe4
    '';
  };
```

untested

nobe4 replied:
kinda done that in 350845c

nobe4 committed:
chore(devenv): simplify

nobe4 committed:
fix(accents): fix the LY_AC and various conditions

nobe4 committed:
fix(accents): A is for acute, G is for grave

nobe4 committed:
feat(accent): support shifted accented characters

Thanks to @sigprof, I was able to implement a working change to get
lowercased and uppercased accented characters.

- On leader-key trigger, the current mod is checked and stored in a global
  variable.
- During the leader sequence checks, if the leader_shifted is on, we
  tap_code16 with S().

This means that for an accented character, the _leader_ key must be
shifted, not any of the leader sequence keys. This changes slightly from
how Espanso was doing it (e.g. `,Eg` now has to be `<S+Leader>eg`) but
it's as good.

nobe4 committed:
feat(ferris): make the accents work on the ferris as well

Somehow flashing again inverted the hands of the keyboard, will have to
fix that later.

nobe4 committed:
fix(ferris): update config and devenv to correctly flash
